jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/SpansToDependencyLinks.java

```diff
@@ -35,26 +38,102 @@
      */
     @Override
     public Iterable<Dependency> call(Iterable<Span> spans) throws Exception {
-        Map<Long, Span> spanMap = new HashMap<>();
+        Map<Long, Set<Span>> spanMap = new LinkedHashMap<>();
         for (Span span: spans) {
-            spanMap.put(span.getSpanId(), span);
+            Set<Span> sharedSpans = spanMap.get(span.getSpanId());
```

Comment:
Instead of maintaining the list of separate spans for the same id, could the subsequent span information be merged into the first span?

Reply:
I think the subsequent implementation would return to the simpler approach before if a merge approach was used.

Reply:
it could but currently there is nothing to merge, we query only ids and service name from process.

The proper implementation could merge here.

Reply:
Think it would be better to query tags aswell, and only store the single/first client span here, and then only process server spans in the loop below - as we should only be interested in relationships between client and server spans. If necessary the loop can then check for client spans with the same span id first (if using shared span) and then parent id second.

Reply:
I would like to have that as a separate PR. We should consider to what complexity do we want to go with this as there will be a different data pipeline where this logic won't be necessary. 

That said, this fixes a lot of use cases and does not add too much logic.